lib/do_profile/profilecommon.js

```diff
@@ -306,6 +307,15 @@ function profileCmonCertgen(opts, cb) {
         function generateAndSignCert(arg, next) {
             var certSignKey = arg.certSignKey;
 
+            if (certSignKey.type === 'ed25519') {
+                console.log(wordwrap('\nThe provided Triton profile: "' +
+                    opts.name + '" is using an ED25519 key which can not be ' +
+                    'used for generating CMON certificates until EDDSA is ' +
+                    'adopted by the CA/Browser Forum.\n'));
```

Comment:
This should mention that ECDSA (only, because RSA can cause performance issues with CMON) should be used instead, the same way you have it for Docker.

And as before:

> Use `--keyId` to select an ECDSA for signing your CMON certificate.

Reply:
I feel like this text is a little misleading -- it's not really the CA/Browser Forum that's the problem here, it's Go's TLS library. The RFCs for x509 and TLS to add support for EdDSA to everything have been published for a while, and OpenSSL and NSS support EdDSA certificates today. The CA/B doesn't tell them what to support, it just decides whether public CAs will start issuing certs using them. We're not really waiting on that: we don't need these signed by a public CA. We just need the TLS lib used by Docker and Prometheus to support it.

Reply:
And actually based on https://github.com/golang/go/issues/25355 it looks like EdDSA cert support made it into Go 1.13, so it might actually just work with one or both of these if built with a new enough Go. When did we last have a try to see if it worked?

Reply:
```
[root@2cb2c281-7c75-47ee-83ac-0dd8e90b0b61 (gps-1:cmon0) ~]# /opt/triton/cmon/build/node/bin/node 
> process.versions.openssl
'1.0.2r'

[root@5b992e40-cf50-4f83-a06e-a6cde68fa05c (gps-1:docker0) ~]# /opt/smartdc/docker/build/node/bin/node 
> process.versions.openssl
'1.0.2r'
```
This might actually be the real stumbling block here at the moment, not the client-side at all if a new enough Go is being used. OpenSSL's support for EdDSA certificates and using it for TLS auth was only added in the 1.1 release series, not 1.0.x

Reply:
> This should mention that ECDSA (only, because RSA can cause performance issues with CMON) should be used instead, the same way you have it for Docker.

I noticed on line 332 it is generating an ECDSA private key:

```
/*
 * Unlike Docker, where choosing ecdsa is somewhat arbitrary,
 * CMON has severe performance issues with RSA at load. Using
 * ecdsa avoids this.
 */

sslKey = sshpk.generatePrivateKey('ecdsa');
```

Does that change anything? Or should we still suggest using a profile with an ECDSA key?

Reply:
@arekinath thanks for the clarification about where the hold up is.

> When did we last have a try to see if it worked?

sdc-docker is using Node.js v6.17.1 with OpenSSL v1.0.2r so it would need to be updated to at least Node 12 to get ED25519 support where we then would also lose MDB support . We were hesitant to do that but it wasn't off the table.  I was under the impression that Docker and Prometheus (both client and server) both currently lacked support for eddsa certificates. I'll see if I can confirm/deny.

Reply:
@travispaul for the CMON key, we do hard code that it’s an ecdsa key, but I actually don’t know how much performance impact there is if the signing key is RSA. It’s my understanding that it needs to verify both the certificate and the signing key, so if thr signing key rsa, that can still be a problem. Maybe @arekinath can chime in here if I’m mistaken.

@arekinath We’ve had a hard time initially getting node 12 or later compiled via sdcnode. I ran across a forum post somewhere about someone asking about it and the reply was that certs currently can’t get issued for eddsa. I had assumed that was client side as well as server side.